### PR TITLE
cfitsio: add 3.490 + bump libcurl + fix hooks + remove headers install subdir

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **cfitsio/3.490**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/cfitsio/all/conandata.yml [hunk 1]
```diff
@@ -5,14 +5,22 @@ sources:
...
     - patch_file: "patches/windows-use-strtok_s.patch"
       base_path: "source_subfolder"
-    - patch_file: "patches/fix-cmake.patch"
+    - patch_file: "patches/fix-cmake-3.470.patch"
       base_path: "source_subfolder"
   "3.480":
     - patch_file: "patches/windows-use-strtok_s.patch"
       base_path: "source_subfolder"
-    - patch_file: "patches/fix-cmake.patch"
+    - patch_file: "patches/fix-cmake-3.470.patch"
```
Comment (prince-chrismc) [inline recipes/cfitsio/all/conandata.yml:20]: Looks odd 👀

**Review by prince-chrismc — APPROVED** [review]
Nice job backporting new settings

File: recipes/cfitsio/all/conandata.yml (hunk 1 above)
Comment (madebr) [reply recipes/cfitsio/all/conandata.yml:20]: Looks weird idd, but is understandable.

File: recipes/cfitsio/all/conanfile.py [hunk 2]
```diff
@@ -94,7 +93,7 @@ def _configure_cmake(self):
         self._cmake.definitions["CFITSIO_USE_SSSE3"] = self.options.get_safe("simd_intrinsics") == "ssse3"
         if self.settings.os != "Windows":
             self._cmake.definitions["CFITSIO_USE_BZIP2"] = self.options.with_bzip2
-            self._cmake.definitions["CFITSIO_USE_CURL"] = self.options.with_curl
+            self._cmake.definitions["UseCurl"] = self.options.with_curl
```
Comment (madebr) [inline recipes/cfitsio/all/conanfile.py:96]: Why did you rename this option?
To avoid the change of mixing up an *official* cfitio option and a custom, patched-in, option?

**Review by madebr — APPROVED** [review]

File: recipes/cfitsio/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/cfitsio/all/conanfile.py:96]: yes
